src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -213,11 +223,11 @@ public string Name
             {
                 try
                 {
-                    return Path.GetFileName(this.FilePath);
+                    return Path.GetFileName(Key.Moniker);
```

Comment:
Yeah, using FilePath vs. the Moniker as a side-channel to communicate that seems very unexpected. If anything, it needs to be commented heavily to understand what's going on here.